### PR TITLE
update SDK dependencies, add OpenSSL

### LINKED ISSUE #50: update cargo-deny to 0.9.1+
We're vending cargo-deny in the SDK now, but it's a fairly old version. We should update to the most recent release and make the necessary changes to `deny.toml` in the OS packages.

### LINKED ISSUE #58: Update Rust to 1.56
https://blog.rust-lang.org/2021/07/29/Rust-1.54.0.html
https://blog.rust-lang.org/2021/09/09/Rust-1.55.0.html
https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**
Fixes #50
Fixes #58 

**Description of changes:**
Refresh all dependencies, including the long-awaited `cargo-deny` update.

Add openssl to the musl sysroot, to be used in places where we're building Rust-based Kubernetes clients.

**Testing done:**
Built Bottlerocket with the updated SDK. Verified that I could create a static Rust binary that linked with OpenSSL by way of `kube-rs` and the `openssl` crate.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -381,7 +381,7 @@ FROM sdk-libc as sdk-go
 
 ARG ARCH
 ARG TARGET="${ARCH}-bottlerocket-linux-gnu"
-ARG GOVER="1.16.5"
+ARG GOVER="1.16.8"
```
Comment (jpculp) [inline Dockerfile]: It looks like a new security release dropped. Should we bump to 1.16.9 instead?

**Review by jpculp — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]
🎉

File: Dockerfile [hunk 2]
```diff
@@ -506,9 +506,9 @@ RUN \
   mkdir -p /usr/libexec/tools /usr/share/licenses/govmomi && \
   chown -R builder:builder /usr/libexec/tools /usr/share/licenses/govmomi
 
-ARG GOVMOMIVER="0.26.0"
-ARG GOVMOMISHORTCOMMIT="34586b6"
-ARG GOVMOMIDATE="2021-06-03T19:03:25Z"
+ARG GOVMOMIVER="0.26.1"
```
Comment (tjkirch) [inline Dockerfile]: 0.27.0 just came out!

**Review by tjkirch — APPROVED** [review]

**Review by jpculp — APPROVED** [review]

**Review by tjkirch — APPROVED** [review]
LGTM; new Rust is coming out in a few days, may be nice to wait for that.

**Review by webern — APPROVED** [review]

**Review by etungsten — APPROVED** [review]

**Pull request merged**